src/action.js

```diff
@@ -0,0 +1,182 @@
+import * as os from "os";
+
+import * as github from "@actions/github";
+import * as toolCache from "@actions/tool-cache";
+import * as core from "@actions/core";
+import * as exec from "@actions/exec";
+
+// Define consts
+
+export const gruntworkOrg = "gruntwork-io";
+export const patcherRepo = "patcher-cli";
+export const patcherVersion = "v0.4.3";
+
+export const reportCommand = "report";
+export const updateCommand = "update";
+
+export const nonInteractiveFlag = "--non-interactive"
+export const noColorFlag = "--no-color"
+export const skipContainerFlag = "--skip-container-runtime"
+
+function osPlatform() {
```

Comment:
Consider updating this function to something like: `isSupportedPlatform(platform:string)`. This will help make the function testable. 

This function will then return a `boolean` if it's not a supported platform and the code calling this function can determine what to do with that (either throw an `UnsupportedPlatform` error or call `core.setFailed` like you are doing here)